### PR TITLE
[SSR] Removed call of requestAnimationFrame outside of function

### LINKED ISSUE #842: Error building gatsby.js, requestAnimationFrame is not defined
```console
failed Building static HTML for pages - 3.096s

 ERROR #95313

Building static HTML failed

See our docs page for more info on this error: https://gatsby.dev/debug-html


  478 |
  479 | let $ea7b4dd250709a616043c0e9cf5f62a$var$fixTs = false;
> 480 | requestAnimationFrame(function (t) {
      | ^
  481 |   $ea7b4dd250709a616043c0e9cf5f62a$var$fixTs = t > 1e12 !== $ea7b4dd250709a616043c0e9cf5f62a$var$getTime() > 1e12;
  482 | });
  483 |


  WebpackError: ReferenceError: requestAnimationFrame is not defined

  - module.js:480
    node_modules/@react-stately/virtualizer/dist/module.js:480:1

  - module.js:1
    node_modules/@react-stately/layout/dist/module.js:1:1

  - module.js:1
    node_modules/@react-spectrum/listbox/dist/module.js:1:1

  - module.js:1
    node_modules/@adobe/react-spectrum/dist/module.js:1:1

  - index.js:1
    src/pages/index.js:1:1
```

### PR DESCRIPTION
**Pull request by fezproof**
This allows the base library of @adobe/react-spectrum to used in Gatsby

Closes #842

## ✅ Pull Request Checklist:

- [X] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [X] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [X] Filled out test instructions.
- [X] Updated documentation (if it already exists for this component).
- [X] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:
- Create a basic hello-world gatsby app
- Add this snippet to your app
```javascript
<Provider theme={defaultTheme}>
  <Flex direction="column" width="size-2000" gap="size-100">
      <View>this is a thing</View>
      <View>this is a thing</View>
      <View>this is a thing</View>
      <View>this is a thing</View>
    </Flex>
</Provider>
```
- Build!

Without this change, the gatsby build will try to run rAF as it is outside of a actually hook function that would normally be called in the render loop. This caused the error in #835. With this simple change, functionality is essentially the same, but it will now compile in a Gatsby enviroment.
The only real loss is that the check will happen every animation frame, but it is a semi-cheap comparison so it shouldn't effect any actual animations.
See https://developers.google.com/web/updates/2012/05/requestAnimationFrame-API-now-with-sub-millisecond-precision#feature_detection to show the change shouldn't make any reasonable difference to the code.

### CODE REVIEW

File: packages/@react-stately/virtualizer/src/tween.ts [hunk 1]
```diff
@@ -42,7 +36,7 @@ export function tween(begin, end, duration, ease, fn): CancelablePromise<void> {
     raf_id = requestAnimationFrame(function run(t) {
       // if we're using a high res timer, make sure timestamp is not the old epoch-based value.
       // http://updates.html5rocks.com/2012/05/requestAnimationFrame-API-now-with-sub-millisecond-precision
-      if (fixTs) {
+      if (t < 1e12) {
```
Comment (devongovett) [inline packages/@react-stately/virtualizer/src/tween.ts:39]: Actually, I think this is the opposite of what we need? If the time is less than 1e12 then it's a high res timer already. I think we want `t > 1e12` here instead?

Comment (devongovett) [reply packages/@react-stately/virtualizer/src/tween.ts:39]: And also we might need to check the resolution of `performance.now()` as the above check does. e.g. in safari `performance.now()` is less than 1e12 (I think after the spectre/meltdown attacks).

Maybe we should just put the check above in an if statement to check if `requestAnimationFrame` even exists?

Comment (fezproof) [reply packages/@react-stately/virtualizer/src/tween.ts:39]: I will experiment with this and get back to you!

Comment (fezproof) [reply packages/@react-stately/virtualizer/src/tween.ts]: Finished testing, doing the compare inside the animation seems to be the only option :(
I have updated the PR + added some `const`s to ease my conscience 😁

**Comment by devongovett** [comment]
Merged manually

**Pull request closed without merging**